src/plugins/intel_gpu/include/intel_gpu/primitives/primitive.hpp

```diff
@@ -260,6 +265,24 @@ struct primitive {
     }
 
 protected:
+    // Fill empty padding when current output paddings less than num_outputs
+    virtual std::vector<padding> update_output_paddings(const std::vector<padding>& paddings, const size_t num_outputs) const {
+        std::vector<padding> updated_paddings(paddings);
+        if (num_outputs > updated_paddings.size()) {
+            updated_paddings.push_back(padding());
+        }
+        return updated_paddings;
+    }
+
+    // Fill empty data_types when current data_types less than num_outputs
+    virtual std::vector<optional_data_type> update_output_data_types(const std::vector<optional_data_type>& data_types, const size_t num_outputs) const {
+        std::vector<optional_data_type> update_data_types(data_types);
+        if (num_outputs > update_data_types.size()) {
+            update_data_types.push_back(optional_data_type());
```

Comment:
I think we can add multiple missing elements in the loop just to make this function more generic

Reply:
I add this function not only loop but also condition, non_max_suppression, arg_max_min and proposal. they use num_outputs for multiple outputs but set only one empty padding for default. 
However, I will change this code for another ways. thanks.